### DISCUSSION TITLE
SO version of winlogbeat doesn't match the official Elastic version?

### DISCUSSION THREAD

**Discussion by abesinger** [Security-Onion-Solutions/securityonion#5229] [Unsupported Versions] [answered]
The [SO documentation on downloading winlogbeat](https://docs.securityonion.net/en/2.3/beats.html#winlogbeat)  states:
>Please note that the hyperlink simply points to the standard Winlogbeat download from the Elastic site.

However, comparing what I download from my SO install  ( winlogbeat 7.13.4, SO version 2.3.61) with the same version manually downloaded from the Elastic site [here] (https://www.elastic.co/downloads/past-releases/winlogbeat-7-13-4), they are not identical. And I don't mean just hashes of the downloads -- I unpacked the .msi files and compared files.  There are a bunch of differences.  (The READMEs  both state the same version number).

So, here are my questions:

1. Is the documentation correct or not (it appears to be incorrect, at a minimum the download link does not actually point to an ES page)? 
2. Are there any functional differences between the SO distributed version and the ES version (from the example config files, it appears that there might be)?

I ask because I have to give sysadmins instructions on installing winlogbeat on their systems.  While I _could_ download the file from SecurityOnion, it's a bit less complicated to just give them a link to the offical ES download page. If the official version is the functional equivalent, I'd rather do that.

**Comment by abesinger** [comment, marked as answer]
I retract my question.  I was not looking at the OSS version of winlogbeat from the Elastic site.  Now that I've found the right version, the SO and Elastic .msi files are identical.

**Reply by defensivedepth** [reply]
One of the big differences between the OSS & non-OSS is that the non-OSS has pre-configured local parsers for some windows eventlogs including sysmon logs. Security Onion parses all of that by default, so its easier to just point people to the OSS version.

**Reply by abesinger** [reply]
That's actually really interesting to know (at least for me :-).  Can you tell me, is the flavor of [the source available on github](https://github.com/elastic/beats/tree/master/winlogbeat) the OSS version? ('d think so but there's nothing there that says either way.

I did a deep dive last week through all the SO ingest pipelines and the winlogbeat source because I wanted to understand clearly how fieldnames are mapped (e.g. what gets mapped to winlog.*, what gets mapped to winlog.event_data.*, etc),  and how compliant it is with the ECS.  If what I was looking at isn't the same as what SO is using, I'm going to have to take another look.

FYi, the reason that I did that deep dive is that I was working on building an ingest pipeline to remap the output from the Windows NXLog CE agent to match what winlogbeat produces). I came to the conclusion that I can't do it properly with NXLog CE -- it would require features that are only available in the $$ version.

**Reply by defensivedepth** [reply]
RE: code on github - not sure tbh. Here is where the SO ingest pipelines start for beats data - https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/files/ingest/beats.common

If you can do some normalization to the NXLog data, you can point it at the SO pipelines - we do that for Winlogbeat, Osquery & Wazuh. Some basic normalization, and then all of them point to the same WEL or Sysmon pipeline - here is the osquery normalization: https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/files/ingest/osquery.query_result#L13

**Reply by abesinger** [reply]
Thanks.  Normalizing NXLog data is exactly what I set out to do, but the problem is that there's no way to properly do it with the CE version  -- it flattens the event_data and user_data fields in with the general log fields.  If you want to preserve them as sub-fields (the same way that winlogbeat does), it's a configuration option only available in the commercial version of NXLog. Also, you don't get SSIDs resolution with the CE version.

I spent a loooooong time looking it before I came to that conclusion. Fortunately I'm told that we don't have that many machines running Windows NXLog, so it won't be that difficult to go with winlogbeat.  And then normalization won't be an issue. :-)

**Reply by defensivedepth** [reply]
Yep agreed!
